src/python/gudhi/representations/archipelago.py

```diff
@@ -0,0 +1,119 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Martin Royer
+
+import copy
+
+import numpy as np
+import pandas as pd
+
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from gudhi.representations.vector_methods import Atol, TopologicalVector, BettiCurve
+
+
+class Archipelago(BaseEstimator, TransformerMixin):
+    """
+    Transformer that dictionary-wraps persistence diagram vectorizers, i.e. objects from gudhi.representations.vector_methods.
+    One provides persistence diagram vectorizers (by way of either `island` or `island_dict`), and the Archipelago object will
+    |fit on| and |transform = vectorize| list or series of persistence diagrams (in pandas format).
+    The object is sklearn-API consistent.
```

Comment:
Isn't the pandas format controlled by `transform_output` in scikit-learn?

Reply:
It is, but that feature I have not implemented: at the moment I force the output in pandas format. Will look into that and if it's easy I will implement it.

Reply:
Actually it is worked out within the TransformerMixin of sklearn, so I branched it up now (not sure if very safe though?) and it nicely gets rid of the annoying pandas dependency.

Reply:
Since the number (and names) of features out of Archipelago can only be known at `transform` run time, I change a class attribute accordingly, and I'm not sure if it is very safe...

Reply:
> Since the number (and names) of features out of Archipelago can only be known at `transform` run time,

I see. It could be computed at `fit` time, which would better match the sklearn philosophy, but that wouldn't be convenient. I have no idea how bad it is, if it can interact strangely with cloning or something. For now, I don't see any obvious problem...